webapp/sources/rudder/rudder-core/src/main/scala/com/normation/cfclerk/domain/VariableAndSectionSpec.scala

```diff
@@ -437,3 +436,20 @@ object DisplayPriority {
     }
   }
 }
+
+sealed  trait ReportingLogic
+
+object ReportingLogic {
+
+  final case object WorstReport extends ReportingLogic
+  final case object SumReport extends ReportingLogic
+  final case class FocusReport(component : String) extends ReportingLogic
+  def apply(value : String) : PureResult[ReportingLogic] = {
+    value match {
+      case "worst" => Right(WorstReport)
+      case "sum" => Right(SumReport)
+      case s"focus:${a}" => Right(FocusReport(a))
+      case _ => Left(Unexpected(s"Value '${value}' is not a valid reporting composition rule."))
+    }
+  }
+}
```

Comment:
I would prefer that each `ReportingLogic` has a `name` attribute, and that name is used everywhere when the string is needed. If we want to ensure consistency. 
So something like: 
```
sealed  trait ReportingLogic { def name: Sring}
final case object WorstReport extends ReportingLogic { val name = "worst" }
final case class FocusReport(component : String) extends ReportingLogic { val name = "focus:${component}" }
etc
```